docs/l3/api_flow.md

```diff
@@ -0,0 +1,413 @@
+}
+```
+
+Member A receives a transaction id for the `demand_a` creation transaction, which they can use to get the details of the transaction using `GET /v1/demandA/{demandAId}/creation/{creationId}`. They can also use `GET /v1/transaction` to get all transactions of any type. 
+
+#### GET /v1/demandA/{demandAId}/creation/{creationId}
+
+Inputting the `demand_a`'s identifier as: `18f832d2-162f-486c-bbc6-d4c436df0d63`
+
+Inputting the `demand_a`'s creation ID as: `8c783421-eb61-4bdb-b719-aee945d02121`
+
+Response:
+```
+{
+  "id": "8c783421-eb61-4bdb-b719-aee945d02121",
+  "state": "finalised",
+  "localId": "18f832d2-162f-486c-bbc6-d4c436df0d63",
+  "apiType": "demand_a",
+  "transactionType": "creation",
+  "submittedAt": "2023-05-09T15:04:19.422Z",
+  "updatedAt": "2023-05-09T15:04:37.227Z"
+}
+```
+
+#### GET /v1/transaction
+
+Inputting the apiType as: `demand_a`
+
+Inputting the status as: `finalised`
+
+Response:
+```
+[
+  {
+    "id": "8c783421-eb61-4bdb-b719-aee945d02121",
+    "state": "finalised",
+    "localId": "18f832d2-162f-486c-bbc6-d4c436df0d63",
+    "apiType": "demand_a",
+    "transactionType": "creation",
+    "submittedAt": "2023-05-09T15:04:19.422Z",
+    "updatedAt": "2023-05-09T15:04:37.227Z"
+  }
+]
+```
+
+Enhancement: The local id changes to `pending` at this point. For example, with Member A:
+```
+{
+  "id": "18f832d2-162f-486c-bbc6-d4c436df0d63",
+  "owner": "Member A",
+  "state": "pending",
+  "parametersAttachmentId": "78b6a81c-5ef3-4549-82a9-e9459ca0cc93",
+  "createdAt": "2023-05-09T14:58:17.408Z",
+  "updatedAt": "2023-05-09T14:58:17.408Z"
+}
+```
+
+The indexers on Member B and Optimiser's dscp-matchmaker-api will process the new `demand_a` when the block containing it is finalised, and the new `demand_a` will be visible to them using `GET /v1/demandA`.
+
+#### Member B
+Inputting the `updatedSince` as: `2023-05-09T14:04:37.227Z`
+
+Response:
+```
+[
+  {
+    "id": "d51277a0-1e23-4d09-aceb-c10ecc147560",
+    "owner": "Member A",
+    "state": "created",
+    "parametersAttachmentId": "da003d33-3f0f-4c18-9ccb-5d122a7176ff",
+    "createdAt": "2023-05-09T15:04:37.279Z",
+    "updatedAt": "2023-05-09T15:04:37.279Z"
+  }
+]
+```
+
+#### Optimiser
+
+Inputting the `updatedSince` as: `2023-05-09T14:04:37.227Z`
+
+Response:
+```
+[
+  {
+    "id": "d51277a0-1e23-4d09-aceb-c10ecc147560",
+    "owner": "Member A",
+    "state": "created",
+    "parametersAttachmentId": "da003d33-3f0f-4c18-9ccb-5d122a7176ff",
+    "createdAt": "2023-05-09T15:04:37.279Z",
+    "updatedAt": "2023-05-09T15:04:37.279Z"
+  }
+]
+```
+
+Member B creates `demand_b` in a similar manner to creating `demand_a`, including the parameters file for `demand_b`.
+
+#### POST /v1/attachment
+
+Response:
+```
+{
+  "id": "18d3b77a-ded6-4fb5-ae16-edd8c1ba1a92",
+  "filename": "attachment_capacity.json",
+  "size": 1304,
+  "createdAt": "2023-05-09T15:32:43.664Z"
+}
+```
+
+#### POST request to /v1/demandB
+
+Response`:
```

Comment:
got an extra ` typo

Reply:
which messes up the formatting of the file after this point

Reply:
```suggestion
Response:
```